model/DataQuery.php

```diff
@@ -702,6 +705,15 @@ public function applyRelation($relation) {
 				$this->query->addLeftJoin($componentBaseClass,
 					"\"$relationTable\".\"$componentField\" = \"$componentBaseClass\".\"ID\"");
 				}
+				else {
+					$alias = uniqid($componentBaseClass);
+					$this->query->addLeftJoin(
```

Comment:
it's not entirely unitelligible because it would be somthing like `SiteTree4b3403665fea6`